src/LuaTransformer.ts

```diff
@@ -3466,6 +3473,10 @@ export class LuaTransformer {
         return tstl.createTableExpression(properties, expression);
     }
 
+    public transformOmittedExpression(node: ts.OmittedExpression): ExpressionVisitResult {
+        return tstl.createNilLiteral(node);
```

Comment:
I think it's not really good to always return `nil` there. `nil` is valid only in one context it could be used (array literals), but it's not in other (destructuring).
I think it would be better to decide it in the visitor (using `node.parent`) and use it in `transformArrayBindingElement`.

Reply:
This neatens things up nicely, I've added `isWithinLiteralAssignmentStatement` which can be used to check if a node is being used in a binding assignment expression. This should work with nested binding patterns too once those are supported.

I've called elements of an array on the left hand side of an assignment statement "ArrayBindingExpressions" since TS refers to them as Expressions and they seem to be a super-set of ArrayBindingElements with PropertyAccessExpressions being another thing they could be.